components/PronunciationChecker/index.tsx

```
@@ -0,0 +1,207 @@
+    setOverallScore(averageScore);
+    setHighlightWords(newHighlightWords);
+    setHighlightColors(newHighlightColors);
+
+    // Gọi callback nếu có
+    if (onScoreChange) {
+      onScoreChange(averageScore);
+    }
+  };
+
+  // Bắt đầu nghe
+  const startListening = useCallback(() => {
```

ESLint

[error] React Hook "useCallback" is called conditionally. React Hooks must be called in the exact same order in every component render.

(react-hooks/rules-of-hooks)